From 49823e6fcfb17b549e20f6867e0612b41351df96 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 13 Sep 2021 11:16:20 +0300
Subject: [PATCH] [SW-55780] gaudi2: pci: enabling pci axi drain

Enabling both hbw and lbw drain in the pci block, so if the NOC is
stuck the pci trasactions will continue to flow and the pci block
will continue to function.

Change-Id: Ic3c22d2aee615cd6e89d4c553ddba3d621018c9d
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 76d00ea..5a8d412 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -727,3 +727,17 @@
 	regmap_writel(AXI_DRAIN_TIMEOUT, mmPSOC_GLOBAL_CONF_AXI_DRAIN_TIMEOUT);
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_AXI_DRAIN_CTRL);
 }
+
+#define HBW_DRAIN_TIMEOUT	0x1000
+#define LBW_DRAIN_TIMEOUT	0x5000
+
+void ppboot_pcie_drain_init(void)
+{
+	regmap_writel(HBW_DRAIN_TIMEOUT, mmPCIE_WRAP_HBW_DRAIN_TIMEOUT);
+	regmap_writel(1, mmPCIE_WRAP_HBW_DRAIN_CFG);
+	regmap_writel(1, mmPCIE_WRAP_HBW_DRAIN_STAMP);
+
+	regmap_writel(LBW_DRAIN_TIMEOUT, mmPCIE_WRAP_LBW_DRAIN_TIMEOUT);
+	regmap_writel(1, mmPCIE_WRAP_LBW_DRAIN_CFG);
+	regmap_writel(1, mmPCIE_WRAP_LBW_DRAIN_STAMP);
+}
diff --git a/include/preboot.h b/include/preboot.h
index 8b09c35..b69a9ae 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -232,6 +232,7 @@
 
 void ppboot_init_stack_mon_dev_specific(void);
 void ppboot_check_stack_mon_dev_specific(void);
+void ppboot_pcie_drain_init(void);
 
 void preboot_main(void);
 
